src/storage/engine/cursor_builder.rs

```diff
@@ -0,0 +1,91 @@
+// Copyright 2018 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use raftstore::store::engine::IterOption;
+use storage::engine::Result;
+use storage::{CfName, Cursor, ScanMode, Snapshot};
+
+/// A handy utility to build a snapshot cursor according to various configurations.
+pub struct CursorBuilder<'a, S: 'a + Snapshot> {
```

Comment:
how about moving `cursor` implementation to this mod and name this mod `cursor`?

Reply:
Good idea!